tests/neg-strict/unchecked-patterns.scala

```diff
@@ -1,13 +1,13 @@
 object Test {
 
-  val (y1: Some[Int] @unchecked) = Some(1): Option[Int]  // OK
+  val (y1: Some[Int]) = Some(1): Option[Int] @unchecked  // OK
   val y2: Some[Int] @unchecked = Some(1): Option[Int]    // error
 
   val x :: xs = List(1, 2, 3)        // error
   val (1, c) = (1, 2)                // error
   val 1 *: cs = 1 *: Tuple()         // error
 
-  val (_: Int | _: Any) = ??? : Any  // error
+  val (_: Int | _: AnyRef) = ??? : AnyRef  // error
```

Comment:
Why do you need this change?

Reply:
This is due to the parametric top, we cannot match `Any`.

Reply:
Ok, that's what I guessed :) Thanks